packages/amplify-cli-core/src/utils/packageManager.ts

```diff
@@ -27,58 +27,41 @@ export interface PackageManager {
 
 const isWindows = process.platform === 'win32';
 
-export class NpmPackageManager implements PackageManager {
-  readonly packageManager: PackageManagerType = 'npm';
+class NpmPackageManager implements PackageManager {
+  readonly packageManager = 'npm';
   readonly displayValue = 'NPM';
-  lockFile;
-  executable;
-  version?: SemVer;
-
-  constructor() {
-    this.lockFile = 'package-lock.json';
-    this.executable = isWindows ? 'npm.cmd' : 'npm';
-  }
+  readonly executable = isWindows ? 'npm.cmd' : 'npm';
+  readonly lockFile = 'package-lock.json';
```

Comment:
these props need to be readonly in the interface as well so TS will enforce it in other places

Reply:
updated